packages/components/table/src/composables/use-scrollbar.ts

```diff
@@ -0,0 +1,32 @@
+import { ref } from 'vue'
+import { isNumber } from '@element-plus/utils'
+
+export const useScrollbar = () => {
+  const scrollBarRef = ref()
+
+  const scrollTo = ({ top, left }: { top?: number; left?: number }) => {
+    setScrollTop(top)
+    setScrollLeft(left)
+  }
+
+  const setScrollTop = (top?: number) => {
+    const scrollbar = scrollBarRef.value
+    if (scrollbar && isNumber(top)) {
+      scrollbar.setScrollTop(top)
+    }
+  }
+
+  const setScrollLeft = (left?: number) => {
+    const scrollbar = scrollBarRef.value
+    if (scrollbar && isNumber(left)) {
+      scrollbar.setScrollLeft(left)
+    }
+  }
```

Comment:
It seems to me that these method are quite similar, can we extract the common part to avoid duplication?

```ts
const setScrollPosition = (position: 'Top' | 'Left', val?: number) => {
  const scrollbar = scrollBarRef.value
    if (scrollbar && isNumber(val) && ['Top', 'Left'].includes(position)) {
      scrollbar[`setScroll${position}`](left)
    }
}

const setScrollTop = (val?: number) => setScrollPosition('Top', val);
const setScrollLeft = (val?: number) => setScrollPosition('Left', val);

```